src/lender/adapters/mkr/clerk.sol

```diff
@@ -0,0 +1,243 @@
+        if (amountDAI > mgr.cdptab()) {
+            repayDAI = mgr.cdptab();
+        }
+        // get DAI from reserve
+        reserve.payout(repayDAI);
+        // repay cdp debt
+        dai.approve(address(mgr), repayDAI);
+        mgr.wipe(repayDAI); 
+
+        // harvest junior interest & burn surplus drop
+        harvest();        
+    }
+
+    // harvest junior profit
+    function harvest() public active {
+        require((cdpink() > 0), "nothing profit to harvest");
+        uint dropPrice = assessor.calcSeniorTokenPrice();
+        uint lockedCollateralDAI = rmul(cdpink(), dropPrice);
+        // profit => diff between the DAI value of the locked collateral in the cdp & the actual cdp debt including protection buffer
+        uint profitDAI = safeSub(lockedCollateralDAI, calcOvercollAmount(mgr.cdptab()));
+        uint profitDROP = rdiv(profitDAI, dropPrice);
+        // remove profitDROP from the vault & brun them
+        mgr.exit(address(this), profitDROP);
+        collateral.burn(address(this), profitDROP);
+        // decrease the seniorAssetValue by profitDAI -> DROP price stays constant
+        decreaseSeniorAsset(profitDAI);
+    }
+
+    // decrease MKR creditline
+    function sink(uint amountDAI) public auth active {
+        require(remainingCredit() >= amountDAI, "decrease amount too high");
+
+        // creditline amount including required overcollateralization => amount by that the seniorAssetValue should be decreased
+        uint overcollAmountDAI = calcOvercollAmount(amountDAI);
+        emit log_named_uint("clerk", overcollAmountDAI);
+        // protection value for the creditline decrease going to the junior tranche => amount by that the juniorAssetValue should be increased
+        uint protectionDAI = safeSub(overcollAmountDAI, amountDAI);    
+        emit log_named_uint("clerk", protectionDAI);
+        // check if the new creditline would break the pool constraints
+        validate(protectionDAI, 0, 0, overcollAmountDAI);  
+        // increase MKR crediline by amount
+        creditline = safeSub(creditline, amountDAI);
+    }
+
+    // checks if the Maker credit line increase could violate the pool constraints // -> make function pure and call with current pool values approxNav
+    function validate(uint juniorSupplyDAI, uint juniorRedeemDAI, uint seniorSupplyDAI, uint seniorRedeemDAI) internal {
+        require((coordinator.validate(juniorSupplyDAI, juniorRedeemDAI, seniorSupplyDAI, seniorRedeemDAI) == 0), "supply not possible, pool constraints violated");
+    }
+    
+    function decreaseSeniorAsset(uint amountDAI) internal  {
+        uint currenNav = nav.currentNAV();
+        uint newSeniorAsset = coordinator.calcSeniorAssetValue(amountDAI, 0,
+            assessor.calcSeniorAssetValue(assessor.seniorDebt(), assessor.seniorBalance()), reserve.totalBalance(), currenNav);
+        uint newSeniorRatio = coordinator.calcSeniorRatio(newSeniorAsset, currenNav, reserve.totalBalance());
```

Comment:
It will be not required to pass the `newSeniorRatio` to the assessor.changeSeniorAsset. 

I think that should simplify it.